deploy/tests/awstools/test_awstools.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import print_function
+from pprint import pprint
+
+# Do NOT import any firesim code being tested that might open connections to AWS here.
+# Import the firesim code to be tested inside the test functions so that the moto
+# mocks are registered before any possible boto connections are created
+import boto3
+from botocore.stub import Stubber
+from moto import mock_sns
+
+from mock import patch
+import pytest
+from pytest import raises
+
+# In case you put any package-level tests, make sure they use the test credentials too
+pytestmark = pytest.mark.usefixtures("aws_test_credentials")
+
+@pytest.mark.usefixtures("aws_test_credentials")
+class TestSNS(object):
+    """Test functions in awstools that work with SNS"""
+
+    # When we're testing awstools.get_snsname_arn(), we don't also want to be
+    # testing awstools.aws_resource_names(). mainly because we're subjugating
+    # the boto3.client factory (in the second test below) and also because we want to be able to run these
+    # tests while not on a AWS EC2 host and aws_resource_names() assumes that it
+    # can lookup the AWS instance-id via 169.254.169.254 API
+    @patch('awstools.awstools.aws_resource_names', return_value={'snsname': 'Testing'})
+    # register the moto backend for sns
+    @mock_sns
+    def test_get_snsname_arn_sanity(self, aws_res_mock):
+        """Simple call returns properly formed ARN"""
+        # local imports of code-under-test ensure moto has mocks
+        # registered before any possible calls out to AWS
+        from awstools.awstools import get_snsname_arn
+
+        arn = get_snsname_arn()
+
+        client = boto3.client('sns')
+        response = client.get_topic_attributes(TopicArn=arn)
+
+        # output will normally be captured and suppressed but printed
+        # iff the test fails. So, leaving in something that dumps the response
+        # can be useful. See https://docs.pytest.org/en/4.6.x/capture.html
+        pprint(response)
+        assert response['ResponseMetadata']['HTTPStatusCode'] == 200
+        assert response['Attributes']['TopicArn'] == arn
+
+        # check that our mock of aws_resource_names was used
+        aws_res_mock.assert_called_once()
+
+
+    @patch('awstools.awstools.aws_resource_names', return_value={'snsname': 'Testing'})
+    @mock_sns
+    def test_get_snsname_arn_auth_exception_handling(self, aws_res_mock):
+        """AuthorizationError is intercepted and re-raised as AssertionError"""
+        # local imports of code-under-test ensure moto has mocks
+        # registered before any possible calls out to AWS
+        from awstools.awstools import get_snsname_arn
+
+        # create a mock SNS client that returns what we tell it to
+        client = boto3.client('sns')
+        stub = Stubber(client)
```

Comment:
It's probably useful to add a comment here that says something like 
```
# we use botocore.stub.Stubber() here instead of mock.patch because boto dynamically
# creates classes at runtime for it's service-specific client objects.  mock.patch can only patch
# static classes that exist in the python symbol table.  botocore.stub.Stubber() is explicitly created for this
# see: https://github.com/boto/boto3/issues/2485
```